### ISSUE TITLE
The ticker is choppy since MM 2.26

### ISSUE THREAD

**Issue by hesspoint** [jalibu/MMM-Jast#64]
### Platform

Pi 4

### Node.js Version

v18.18.0

### MagicMirror Version

2.26

### Module Version

lastest version

### Description

I have been running the module for a few month w/out issues. Since I updated MM to 2.26 recently the ticker does not run smoothly anymore. It starts stopping ever second or so and then jumps a bit ahead. It is very choppy and nice to look at. Before it has never done that.

### Steps to reproduce

same after every restart or Pi reboot

### config.js

```javascript
module: "MMM-Jast",
			position: "bottom_bar",
			config: {
				currencyStyle: "code", // One of ["code", "symbol", "name"]
				fadeSpeedInSeconds: 120,
				lastUpdateFormat: "HH:mm",
				maxChangeAge: 1 * 24 * 60 * 60 * 1000,
				maxWidth: "100%",
				numberDecimalsPercentages: 1,
				numberDecimalsValues: 2,
				scroll: "horizontal", // One of ["none", "vertical", "horizontal"]
				showColors: true,
				showCurrency: true,
				showChangePercent: true,
				showChangeValue: false,
				showChangeValueCurrency: false,
				showHiddenStocks: false,
				showLastUpdate: false,
				showPortfolioValue: false,
				showPortfolioGrowthPercent: false,
				showPortfolioGrowth: false,
				updateIntervalInSeconds: 300,
				useGrouping: false,
				virtualHorizontalMultiplier: 2,
				stocks: [
					{ name: "Tesla", symbol: "TSLA"},
					{ name: "Amazon", symbol: "AMZN"},
					{ name: "Apple", symbol: "AAPL"},
				]
			}
```

**Comment by jalibu** [comment]
Hi Hannes,
thank you for opening this issue.
I can not really explain why the rendering performance of the module could be related to the new MM version.
The rendering is done by the browser (these are normal CSS transition animations).

Probably, it could be that the general resource consumption is higher due to the new MM version and therefore less is left for rendering the ticker.

I don't have this problem, by the way.
We can wait and see if other affected people get in touch.

**Comment by jalibu** [comment]
@stengelchristian what do you mean with that?
> BUT AV STOCK Module looks exactly the same

**Comment by jalibu** [comment]
Hi @hesspoint  @stengelchristian 

can you please checkout the latest version and share your feedback. I made some changes in the horizontal scrolling

**Comment by stengelchristian** [comment]
> @stengelchristian what do you mean with that?
> 
> > BUT AV STOCK Module looks exactly the same

I was referring to the https://github.com/lavolp3/MMM-AVStock Module, which also has a horizontal scroll. The scrolling motion had the same choppy behavior from time to time.

I just checked out the 2.9.2 commit, and I would say that it is less choppy, occurring less frequently but I will check again tomorrow then it´s updating stock data and displaying more content.

thx for looking into it.

**Comment by KristjanESPERANTO** [comment]
If it does occur again, you could perhaps try the following:

Instead of `npm run start` try `export ELECTRON_ENABLE_GPU=1 && npm run start` to start the mirror.

With MM 2.26.0 the GPU is disabled by default. This could be the reason why it has not been running so smoothly since then.

**Comment by jalibu** [comment]
@sdetweil do you think it's worth thinking about adding this option to enable GPU rendering as an option for your installer scripts?

**Comment by sdetweil** [comment]
@jalibu but it takes more than that. the drivers in /boot/config.txt need to be set correctly, as I understand it

**Comment by jalibu** [comment]
@sdetweil I thought you are the no-limits-scripting-magic-wonder-man 😁

**Comment by sdetweil** [comment]
yes  I can do lots of things.. IF  I know what to do. for the video drivers I do NOT KNOW. seems like black.magic to me.

some drivers impact commands to manipulate display status

I know I can change the driver config on pi 4/5 to have two independent displays , instead of one big display over both monitors..

this is not my area of expertise..  I'll take advice and testing from others.

**Comment by hesspoint** [comment]
> If it does occur again, you could perhaps try the following:
> 
> Instead of `npm run start` try `export ELECTRON_ENABLE_GPU=1 && npm run start` to start the mirror.
> 
> With MM 2.26.0 the GPU is disabled by default. This could be the reason why the horizontal scrolling is not running so smoothly since then.

I have tried that and yes then it and it is much smoother. However it also is not entirely smooth but at least it does not do these weird jumps.

The issue is though that I start my MM via pm2. Not sure if I can somehow enable the GPU there.

Or is there a setting somewhere in the MM config where I can enable GPU?

**Comment by sdetweil** [comment]
@hesspoint out it on the script that pm2 launches.

if using my install script  that is
~/MagicMirror/installers/mm sh

you can also find this thru the pm2 info command

pm2 status will give u the name and number of the managed apps

pm2 info name/number

look for script path

**Comment by BillYovino** [comment]
Pi Zero 2W running Bullseye. I tried everything I could but the scrolling was still too jerky so I reverted back to my older version running on Buster which works fine. The vertical scrolling didn't look bad but I don't like how it behaves. It scrolls down one stock at a time but when it gets to the end of the list it scrolls up through all of them rapidly. A more natural way would just to continuously scroll down one at a time. I tried over-clocking but that yielded strange results and didn't fix the jerky motion.

**Comment by sdetweil** [comment]
you may have to enable the gpu for scrolling to be smooth

in 2.27 we turned it OFF by default to prevent errors MOST users have. 

in the script that launches MM add
export ELECTRON_ENABLE_GPU=1

before the npm start line

note you MAY have to change the /boot/config.txt to.make it work

**Comment by andrewtwist** [comment]
I'm experiencing the same choppy issue.

**Comment by sdetweil** [comment]
@andrewtwist  did you see my post immediately before yours?

**Comment by andrewtwist** [comment]
Yes, I did see your post.  It was still choppy even with your suggestion.  But, now that it has been running for a few hours it seems to be more stable than when first started.

**Comment by sanj926** [comment]
I saw no issues with performing the fix you mentioned within the script that launches MM.  How do I change the boot/config?  What do I change in there?

**Comment by sanj926** [comment]
I saw great improvement after disabling the display of seconds on my clock module!

**Comment by sdetweil** [comment]
awesome. i do mot know what to change in boot to enable the gpu
